Update dependencies and fix import.

davidlehn opened the pull request.

davidlehn committed:
Update actions.

- Test with Node.js 20.x.
- Update action verions.

davidlehn committed:
Use vue3 eslint rules.

davidlehn committed:
Update dependencies.

- Update all dependencies.
- Update quasar import to avoid errors due to no default export.

dlongley reviewed (approved):
LGTM, AFAICT. Thanks!

davidlehn committed:
Remove bedrock override to use quasar UMD bundle.

davidlehn commented:
> LGTM, AFAICT. Thanks!

It may have looked good... it was not.  It didn't work in top-level apps that use bedrock-webpack due to a UMD bundle override.  Removed that and webpack will now use ESM bundle in ESM apps.  I don't think there is an issue with doing this in general these days.  However, that override caused any `quasar` usage to use UMD, and those bundles have some differences.  Changing to ESM bundle can break apps, depending on their usage, so it's a major change.  I put in patches for a few other packages, and others can be updated as needed.

dlongley reviewed (approved):
Bummer it had to be breaking, but it's for the best. Thanks!

davidlehn committed:
Add quasar lint support.

davidlehn committed:
Update dependencies.

davidlehn committed:
Remove `animate.css` `compat` override.

**BREAKING**: Remove `animate.css` `compat` override and use latest
style.  See the [migration guide](https://animate.style/#migration) for
details on updating.

davidlehn committed:
Update engine.node minimum to >=18.

davidlehn committed:
Update lint dependencies.

davidlehn committed:
Update to `quasar@2.14.6`.

davidlehn committed:
Add support for Quasar initialization options.

Using ESM vs UMD for Quasar means plugins are no longer used
automatically or by default. Applications must be initialized with
Quasar plugins it uses. This is currently not automatic and requires
top-level application knowledge of all sub-modules in use and what
plugins they require. The `quasarOptions` object added here can be used
for this purpose and for other Quasar options.

mattcollier commented on CHANGELOG.md:
```diff
@@ -1,5 +1,32 @@
+    and may need to update imports due to differences in the UMD vs ESM quasar
+    bundles.
+  - Change `import Quasar from 'quasar';` to `import {Quasar} from 'quasar';`.
+  - Change `import {default as Quasar} from 'quasar';` to `import {Quasar} from
```
There's some issue with backticks here.

davidlehn replied:
I think it's rendering HTML correctly?  Are newlines not supposed to be used in md backtick blocks?

mattcollier commented on package.json:
```diff
@@ -32,38 +32,21 @@
   },
   "engines": {
-    "node": ">=14"
+    "node": ">=18"
```
I know this does not inform a major release, but does it at least get a mention in the changelog?

davidlehn replied:
It can be mentioned.  What should it say?  There are no direct tests here and I'm not sure what the actual requirement is.  This is mostly to lock to modern runtimes.

mattcollier replied:
Required Node.js 18+.

mattcollier reviewed (approved):
TY

mattcollier commented:
@davidlehn We're running into issues with Quasar `2.14.6` in multiple projects now.  Can you keep moving this forward to a release?  Thank you.

davidlehn committed:
Update changelog.

davidlehn committed:
Update dependencies.

davidlehn committed:
Remove old comment.

davidlehn committed:
Distribute license files.
